ide/editor.structure/test/unit/src/org/netbeans/modules/editor/structure/api/DocumentModelTest.java

```diff
@@ -294,48 +305,70 @@ public void updateModel(DocumentModel.DocumentModelModificationTransaction dtm,
         
     }
     
-    private static class DocumentModelListenerAdapter implements DocumentModelListener {
+    private class DocumentModelListenerAdapter implements DocumentModelListener {
```

Comment:
Please make these classes static again - non static inner classes hold a hard reference to their enclosing object. which can cause suptle memory issues. Yes I know, that this is just test code, but you also added @Override annotations (which is ok), so I'd like this to be reverted.

That is also true for FakeDocumentModelProvider.

Reply:
@matthiasblaesing I reverted the changes.